sauna-app/pages/index.tsx

```diff
@@ -26,12 +34,22 @@ function SaunaPage(props: SaunaData) {
 
 
 //data fetching for pre-rendering
-export async function getStaticProps() {
-  const saunas = await fetchData()
+export async function getStaticProps(): Promise<SaunaProps> {
+  //OPTION 1 TO FETCH DATA:
+  //const saunas = await fetchData()
+  
+  //OPTION 2 TO FETCH DATA USING API
+  const saunas: SaunaData = await fetch('http://localhost:3000/api/saunas')
+    .then((response) => {
+      return response.json();
+    })
+    .then((data) => {
+      return data
+    })
 
   return {
     props: {
-      saunas: saunas.map(s => ({
+      saunas: saunas.saunas.map(s => ({
```

Comment:
Nit: You could maybe destructure the object to have `_id` and `rest`, and then spread the rest-object, and then set the id. I'm not giving (yet) an example here; This is extra if you want  to study more. This code works, and what I'm suggesting here is more for learning :) 

So, two things to google: Object desctucturing (with rest-parameter) and object spreading. And if you have any questions, don't hesitate to ask 😊

Reply:
Sounds like a plan!